.github/workflows/test-unit.yml

```
@@ -31,3 +31,8 @@ jobs:
 
       - name: Run unit tests
         run: make test-unit
+
+      - name: Upload coverage reports to Codecov
+        uses: codecov/codecov-action@v4.0.1
```

Can we pin by hash? latest version is `v4.4.1`

https://github.com/codecov/codecov-action/commit/125fc84a9a348dbcf27191600683ec096ec9021c

```suggestion
        uses: codecov/codecov-action@125fc84a9a348dbcf27191600683ec096ec9021c # v4.4.1
```